validators/types.go

```diff
@@ -0,0 +1,96 @@
+package validators
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/umbracle/fastrlp"
+)
+
+var (
+	ErrMismatchValidatorType    = errors.New("mismatch between validator and validator set")
+	ErrMismatchValidatorSetType = errors.New("mismatch between validator sets")
+	ErrValidatorAlreadyExists   = errors.New("validator already exists in validator set")
+	ErrValidatorNotFound        = errors.New("validator not found in validator set")
+)
+
+type ValidatorType string
+
+const (
+	ECDSAValidatorType ValidatorType = "ecdsa"
+	BLSValidatorType   ValidatorType = "bls"
+)
+
+func (t *ValidatorType) FromString(s string) error {
+	x := ValidatorType(s)
+
+	switch x {
+	case ECDSAValidatorType, BLSValidatorType:
+		*t = x
+
+		return nil
+	default:
+		return fmt.Errorf("invalid validator type: %s", s)
+	}
+}
+
+func NewValidatorFromType(t ValidatorType) Validator {
```

Comment:
Unused ctor

Reply:
Removed in the next PR, a85e43ec5b5186573f08192d7be09d99242748b4